relay-server/src/actors/outcome.rs

```diff
@@ -114,6 +114,9 @@ pub enum Outcome {
     /// The event has been discarded because of invalid data.
     Invalid(DiscardReason),
 
+    /// The event has already been discarded on the client side.
+    ClientDiscard(String),
```

Comment:
I have some concern that SDKs are going to pick different reasons for basically the same concept, or that for one reason or another this ends up being a very high-cardinality column. Is there no chance we can define this upfront like DiscardReason, especially as reason is now untrusted input via this PR?

Reply:
The reasons are typed in snuba. This way we do not need to force relay updates for new types. This was a decision I made with Jan as we feel this is going to cause less issues.